tiledb/sm/query/query.cc

```diff
@@ -2051,5 +2107,12 @@ void Query::reset_coords_markers() {
   }
 }
 
+void Query::finalize_aggregates() {
```

Comment:
- Why is this call separated from the computation step?
- What are the preconditions for this call and how do we check them?

Reply:
This finalization step copies the aggregated data to the user buffers. We don't want to touch the user buffers until we have the aggregate computed. I'm not sure how we can check any preconditions here as this will copy the current value of the aggregate to the user buffers... Note that it could be a valid case to have no results and copy an send an empty aggregate value back to the user.